File: benchmark/results/java-invocation-script/run-004/b-baseline/output/com/metaengine/demo/catalog/services/ProductRepository.java
package com.metaengine.demo.catalog.services;

import com.metaengine.demo.catalog.aggregates.Product;
import java.util.List;

/** ProductRepository persistence service in the catalog domain. */
public class ProductRepository {
    /** Persist a new Product from a partial input. */
    public Product create(Product input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Find a Product by id, or null if not found. */
    public Product findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** List Products up to limit. */
    public List<Product> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Delete a Product by id. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: benchmark/results/ts-multilang/run-002/b-baseline/output/src/domain/catalog/services/ProductRepository.ts
import { Product } from "../aggregates/Product.js";

/** ProductRepository domain service. */
export class ProductRepository {
  create(input: Partial<Product>): Product {
    void input;
    throw new Error("not implemented");
  }

  findById(id: string): Product | null {
    void id;
    throw new Error("not implemented");
  }

  list(limit: number): Product[] {
    void limit;
    throw new Error("not implemented");
  }

  delete(id: string): void {
    void id;
    throw new Error("not implemented");
  }
}
